tests/diagnostics/non-void-return-constructor.slang

```diff
@@ -0,0 +1,35 @@
+//TEST:SIMPLE(filecheck=CHECK):
+
+// Test to catch compilation time errors with return values in a constructor / member functions within a class / struct.
+struct MyType
+{
+    float v;
+ 
+    // A constructor from the callsite's point of view is a function that returns MyType.
+    __init()
+    {
+        // A constructor from the inside the body is treated as a function that modifies `this`.
+        // because we are expected to write:
+        // this.v = 5;
+        // here.
+ 
+        MyType t;
+        t.v = 5;
+        // CHECK: ([[# @LINE+1]]): error 39999: constructor cannot return a value
+        return t;
+    }
```

Comment:
Can we have one more test case that a constructor can call "return" without any value?

Reply:
can I put that in the same file, or create a new one? 
my concern is that no errors will be reported when we have a constructor with a return with no value. so, i can't add a CHECK line.

Reply:
You will need to have a new file for a good test case.

Reply:
Thanks Jay. But based on an offline chat, i ended up adding an extra case in the same file.